videoretrieval/languagemodels/gpt.py

```diff
@@ -16,75 +16,31 @@
 """
 
 from base import BaseLanguageModel
-from transformers import TFOpenAIGPTModel, OpenAIGPTTokenizer
-
-GPT_PAD_TOKEN = 0
+from transformers import TFOpenAIGPTModel, OpenAIGPTTokenizerFast
 
 class OpenAIGPTModel(BaseLanguageModel):
     """An implementation of BaseLanguageModel for the openai-gpt1 model."""
 
-    max_input_length = 37
-    _batch_size = 42
-
-    def __init__(self):
-        self.model = TFOpenAIGPTModel.from_pretrained("openai-gpt")
-        self.tokenizer = OpenAIGPTTokenizer.from_pretrained("openai-gpt")
-
     @property
     def name(self):
         return "openai_gpt"
 
     @property
     def batch_size(self):
-        return self._batch_size
+        return 42
 
     @property
-    def encoded_shape(self):
-        return (self.max_input_length,)
+    def max_input_length(self):
+        return 37
+
+    def get_tokenizer(self):
+        tokenizer = OpenAIGPTTokenizerFast.from_pretrained("openai-gpt")
+        tokenizer.pad_token = " "
+        return tokenizer
+
+    def get_model(self):
+        return TFOpenAIGPTModel.from_pretrained("openai-gpt")
 
     @property
     def contextual_embeddings_shape(self):
         return (37, 768)
```

Comment:
Can you comment on what these numbers correspond to?

A better way is to assign them to variables named carefully and returning those variables. Then, you won't need any comment.

Reply:
+1

Reply:
Updated!